### ISSUE TITLE
Alternate Target and Alternate Destination Marker problem 

### ISSUE THREAD

**Issue by user3247548** [duplicati/duplicati#3966]
- **Duplicati version**: 2.0.4.23_beta_2019-07-14
- **Operating system**: Version	10.0.17134 Build 17134
- **Backend**: Local internal spinning disk or External USB

## Description
I think the concept of Alternate Target and Alternate Destination Marker is a wonderful idea. However, to me there seems to be a bug in the syntax that seems to make it impossible to use this feature. 
I have reviewed
https://www.duplicati.com/articles/Windows-Drive-Letters/

I use Windows 10 and the latest beta Version: 2.0.4.23 (2.0.4.23_beta_2019-07-14) I created the backup after starting from the tray icon (right click open) arriving at the webpage on localhost:8200 webpage interface. On station (2) ‘definition of the Backup destination’ I put the primary folder Path T:\XYZ.
Under Advanced Options by enter the alternate Target as *:\XYZ with or without an additional \ at the end.
Then under Advanced Option I add an alternative destination marker XYZ.txt and then manually create the folders and the text file
I have found:
https://forum.duplicati.com/t/windows-external-hard-disk-recognize-usb-variable-drive-letter/3714/3
And I ensured the magic alternative destination marker file has to be inside the backup folder.
I always receive:
the Warning that is discussed here:
https://forum.duplicati.com/t/option-altern-alternate-target-paths-is-not-supported-and-will-be-ignored/8300
the 6 letters “?altern” were automatically added to my path when I added the two Advanced Option.
In one of the responses Someone else mentions that “?auth-u” is automatically added to their path when they try to use the username and password Advanced Option.
There someone else speculated that could be related to the “file://” string that is added to the path at the left end and then the counting of letters would be inconsistent.
Also these two Advanced Option seem not persistent from one edit to the next.
For example if I edit the filters in station (3) they are still there when I go to station (3) in another editing session. But the two alternative destination related options disappear on the webpage.
If I go to home ==> commandline I was able to see these alternative destination options were added to the Path in a confusing manner as: 
```
T:\Duplicati_10_2019?altern?alternate-target-paths=*%3A%5CDuplicati_10_2019&alternate-destination-marker=Charles_10_2019.txt
```

Perhaps, the experts do not use the graphical input on the webpage and build all commands on the command line and therefore will not notice if the graphical webpage is broken. But I could not find the documentation how the commandline should be used for alternate paths.
By any stretch of imagination I cannot get from the examples given :
--alternate-destination-marker=my_backup_is_here.txt
--alternate-target-paths=*:\my_backup
to the path I see in my GUI-generated paths string.
https://duplicati.readthedocs.io/en/latest/04-using-duplicati-from-the-command-line/ does not mention “altern”

## Steps to reproduce

1. Install duplicati on Win10
2. Start web GUI from Tray icon
3. Build backup instructions from GUI using alternate path

Warning: Option –altern?alternate-target-paths is not supported and will be ignored

!-- For "Expected result", describe what should happen after you run the steps above (i.e. corrected behavior). -->
- There should not be a warning and red tray icon. If I provide a S:\ drive with magic path and *.txt the backup should go there even if I build the original destination on the T:\ drive

Please see forum contributions of quotwodupl for what was tried before submitting the github issue.

**Comment by nanobug** [comment]
In the forum user ts678(moderator) suspects a bug in substring handling.
[https://github.com/duplicati/duplicati/blob/2c86bcc774abc85872f9372e66dd97b90a39b399/Duplicati/Server/webroot/ngax/scripts/services/EditUriBuiltins.js#L273-L276](https://github.com/duplicati/duplicati/blob/2c86bcc774abc85872f9372e66dd97b90a39b399/Duplicati/Server/webroot/ngax/scripts/services/EditUriBuiltins.js#L273-L276)

[https://forum.duplicati.com/t/auth-u-automatically-added-to-destination-path/5298/2](https://forum.duplicati.com/t/auth-u-automatically-added-to-destination-path/5298/2)

**Comment by ts678** [comment]
[Fixed destination folder path](https://github.com/duplicati/duplicati/commit/af0018c10c54f508e869ea34f31ad159c76bcda2)

> Fixed destination folder path displaying the wrong portion of the path when the URI has advanced options.

might have fixed that GUI issue on substring handling. I don't have a nice test case, but you can try it (probably if you test, you should get a newer Canary such as [v2.0.4.34-2.0.4.34_canary_2019-11-05](https://github.com/duplicati/duplicati/releases/tag/v2.0.4.34-2.0.4.34_canary_2019-11-05) to enjoy some bug fixes and hopefully fewer breakages. Canary is pretty raw, but sometimes good stuff.

[v2.0.4.29-2.0.4.29_canary_2019-09-17](https://github.com/duplicati/duplicati/releases/tag/v2.0.4.29-2.0.4.29_canary_2019-09-17)

> Fixed a parsing issue when reading the server path in the UI, thanks @FlyingFox333

> I also can't use the alternate-target option with the web frontend.

IIRC before I got pulled into some other emergency, I was looking into this, and I think the place to put this is on Advanced options on screen 5 Options -- except it's not there to use on old releases. I found:

![image](https://user-images.githubusercontent.com/40771998/68956282-bff13900-0795-11ea-8f30-1f322f46c1e0.png)

sitting around, and I think it worked. If you have another system, or dare try Canary, you could test this.

**Comment by nanobug** [comment]
Used the 2019-11-05 canary and confirm that the 'alternate' option now work in that version.
Thanks a lot.

**Comment by warwickmm** [comment]
I believe this was fixed by pull request #3903 (released in v2.0.4.29).
